### PR TITLE
Possible fix for conflicting scoreboards

### PR DESCRIPTION
**Pull request by pietru2004**
I think that if multiple projects will use the same scoreboard name it may create conflicts so it would be best to add a board name value so each plugin could have own unique scoreboard.

### CODE REVIEW

**Comment by Tigerpanzer02** [comment]
May it would be make more sense to just use for example the first 4 chars of the plugin name that you can get out of the plugin description

**Comment by pietru2004** [comment]
Well with using string in function one plugin could use multiple scoreboards

**Comment by titivermeesch** [comment]
@pietru2004 @Tigerpanzer02 A good idea could be to have a 4 char (don't really care about the length) identifier acting as a prefix and some (random) string to make it truly unique?

This would make it easy to identify to whom a scoreboard belongs as well. This combines both of your ideas.

**Comment by pietru2004** [comment]
hm I rl wasn't knowing how to get plugin name that runs scoreboard func but genning just a unique string seams easier...

**Comment by titivermeesch** [comment]
> hm I rl wasn't knowing how to get plugin name that runs scoreboard func but genning just a unique string seams easier...

Something that could help getting the plugin name would be to setup a scoreboard manager that would require a plugin instance to work. This would require a heavy core change in the system though and I'm not familiar enough with the project yet

**Comment by pietru2004** [comment]
ok I wasn't sure how to gen a string so I searched a bit and implemented It.
So now it should work like if you provide your own string it will use that string and if no string provided then it will gen 6 long random string.

File: pom.xml [hunk 1]
```diff
@@ -43,6 +43,12 @@
             <version>4.0.0</version>
             <scope>provided</scope>
         </dependency>
+        <dependency>
+            <groupId>org.jetbrains</groupId>
+            <artifactId>annotations</artifactId>
+            <version>RELEASE</version>
```
Comment (titivermeesch) [inline pom.xml:49]: This is not how a dependency version is defined. It should probably be something like `16.0.2`

File: src/main/java/me/tigerhix/lib/scoreboard/ScoreboardLib.java [hunk 2]
```diff
@@ -24,25 +27,43 @@ public static void setPluginInstance(Plugin instance) {
     ScoreboardLib.instance = instance;
   }
 
-  public static Scoreboard createScoreboard(Player holder) {
+  public static Scoreboard createScoreboard(Player holder, @Nullable String new_board_name) {
+    String board_name;
+    if (new_board_name!=null && new_board_name.length()>0) {
+      board_name=new_board_name;
+    } else {
+      int leftLimit = 48; // numeral '0'
```
Comment (titivermeesch) [inline src/main/java/me/tigerhix/lib/scoreboard/ScoreboardLib.java:35]: This logic is something that would have a better place in some sort of utility file

**Review by titivermeesch — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (pietru2004) [reply pom.xml:49]: It was generated by Inteligj IDEA automatically. And it works from what I seen.

File: src/main/java/me/tigerhix/lib/scoreboard/ScoreboardLib.java (hunk 2 above)
Comment (pietru2004) [reply src/main/java/me/tigerhix/lib/scoreboard/ScoreboardLib.java:35]: ok I will try to do it :) after Christmas probably.

**Comment by Tigerpanzer02** [comment]
Another integration is made by plugily to avoid it

**Pull request closed without merging**